### DISCUSSION TITLE
How to group elements?

### DISCUSSION THREAD

**Discussion by miasik** [hako-mikan/sd-webui-supermerger#76] [Q&A] [answered]
As I see in the table here [elemental_en.md](https://github.com/hako-mikan/sd-webui-supermerger/blob/main/elemental_en.md) there are six elements within IN00 block:

1. 0.bias
2. 0.weight
3. time_embed.0.weight
4. time_embed.0.bias
5. time_embed.2.weight
6. time_embed.2.bias
 
I'd like to group them in pairs 1+2, 3+4, 5+6 and make some test with these pairs, not single elements.
If I'm right I can describe the 3+4 pair as "IN00:time_embed.0:" and the 5+6 pair as "IN00:time_embed.2:". So, the final string will be "IN00:time_embed.0:,IN00:time_embed.2:". Am I right?
How to describe the 1+2 pair in order to get only 1st and 2nd elements within the pair?

**Comment by miasik** [comment]
As I see, there's something wrong with my understanding.
I haven't managed to compose my string for "Sequential Merge Parameters" using your "IN00:" format, but had some success with the next form:
`input_blocks.0.0:, model.time_embed.0:, model.time_embed.2:`
Using the string above I got a log(partially):

> ['input_blocks.0.0'] [''] model.diffusion_model.input_blocks.0.0.weight 1.0
> ['input_blocks.0.0'] [''] model.diffusion_model.input_blocks.0.0.bias 1.0

> ['model.time_embed.0'] [''] model.diffusion_model.time_embed.0.weight 1.0
> ['model.time_embed.0'] [''] model.diffusion_model.time_embed.0.bias 1.0

> ['model.time_embed.2'] [''] model.diffusion_model.time_embed.2.weight 1.0
> ['model.time_embed.2'] [''] model.diffusion_model.time_embed.2.bias 1.0

And the resulting picture looked as expected, with 3 columns.

I doubt if I'm right. Still need your guide.

**Comment by hako-mikan** [comment, marked as answer]
You are right.
As you shown in your log. element name has long name such as `model.diffusion_model.input_blocks.0.0.weight`
so you can separate each other using long element name.
```
IN00:blocks.0.0:1.0
IN00:time_embed.0:1.0
IN00:time_embed.2:1.0
```
you will get
```
['IN00'] ['blocks.0.0'] model.diffusion_model.input_blocks.0.0.weight 1.0
['IN00'] ['blocks.0.0'] model.diffusion_model.input_blocks.0.0.bias 1.0

['IN00'] ['time_embed.0'] model.diffusion_model.time_embed.0.weight 1.0
['IN00'] ['time_embed.0'] model.diffusion_model.time_embed.0.bias 1.0

['IN00'] ['time_embed.2'] model.diffusion_model.time_embed.2.weight 1.0
['IN00'] ['time_embed.2'] model.diffusion_model.time_embed.2.bias 1
```
